### DISCUSSION TITLE
Why do I use rigid bodies created with Cylinder and Sphere, where Cylinder does not participate in collisions, and rigid bodies created with two Spheres are normal

### DISCUSSION THREAD

**Discussion by 3261564019** [pmndrs/cannon-es#196] [Q&A]
https://github.com/pmndrs/cannon-es/assets/70420794/993e91f0-81c1-4cc6-80ff-4ff6de6a3d24

https://github.com/pmndrs/cannon-es/assets/70420794/8ca1b812-e08c-4a71-a066-778e16277e9f

**Comment by isaac-mason** [comment]
It's hard to say for sure without seeing some code / a codesandbox, but one possible explanation is you're using shapes that don't support collision with each other.

If you're using a Trimesh shape for your environment, your issue would be using unsupported collision pairs. In cannon Cylinder and Trimesh shapes can't collide with each other at present - the collision logic for that pair hasn't been implemented.

You can find a table showing the supported collision shapes here: https://github.com/schteppe/cannon.js

**Reply by isaac-mason** [reply]
I've added that table to the docs website now: https://github.com/pmndrs/cannon-es/pull/197

**Reply by isaac-mason** [reply]
https://pmndrs.github.io/cannon-es/docs/index.html#supported-shape-collision-pairs

**Reply by 3261564019** [reply]
Very useful Thank you.
